src/fem/patch_point_values.hh

```diff
@@ -0,0 +1,250 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ *
+ * @file    patch_point_values.hh
+ * @brief   Store finite element data on the actual patch
+ *          such as shape function values, gradients, Jacobian
+ *          of the mapping from the reference cell etc.
+ * @author  David Flanderka
+ */
+
+#ifndef PATCH_POINT_VALUES_HH_
+#define PATCH_POINT_VALUES_HH_
+
+#include "fem/eigen_tools.hh"
+#include "fem/dh_cell_accessor.hh"
+
+
+template<unsigned int spacedim> class PatchFEValues;
+template<unsigned int spacedim> class ElOp;
+using Scalar = double;
+using Vector = arma::vec3;
+using Tensor = arma::mat33;
+
+
+static const uint INVALID_COLUMN = -1;
+
+
+
+template<unsigned int spacedim = 3>
+class PatchPointValues
+{
+public:
+    /// Default constructor
+    PatchPointValues(uint dim);
+
+    /// Initialize object, set number of columns (quantities) in tables
+    void initialize(uint int_cols);
+
+    /// Reset number of rows (points)
+    inline void reset() {
+        n_points_ = 0;
+        n_elems_ = 0;
+    }
+
+    /// Getter of n_columns_
+    inline uint n_columns() const {
+        return n_columns_;
+    }
+
+    /// Adds the number of columns equal to n_added, returns index of first of them
+    inline uint add_columns(uint n_added) {
+        uint old_size = n_columns_;
+        n_columns_ += n_added;
+        return old_size;
+    }
+
+    /// Register element to patch_point_vals_ table by dimension of element
+    uint register_element(DHCellAccessor cell, uint element_patch_idx);
+
+    ElOp<spacedim> *add_accessor(ElOp<spacedim> *op_accessor);
+
+protected:
+    TableDbl point_vals_;
```

Comment:
Describe. Main properties:
TableInt same number of rows as point_vals_, possibly only indices of elements associated with points.
Global point index on the patch. 

For ElVals and PointVals describe the Ops that does not depend on other and must be initialized during the creation of the PatchPointVlaues instance. ElVals - coords of vertices, PointVals - coords of points.